lib/matplotlib/animation.py

```diff
@@ -903,8 +903,16 @@ def _start(self, *args):
         # First disconnect our draw event handler
         self._fig.canvas.mpl_disconnect(self._first_draw_id)
 
-        # Now do any initial draw
-        self._init_draw()
+        try:
+            # Now do any initial draw
+            self._init_draw()
+        except StopIteration:
+            # we can't start the iteration, it may have already been
+            # exhausted by a previous save, bail.
+            warnings.warn("Can not start iterating the frame data.  This "
+                          "is either due to frames having zero length or "
+                          "being a generator which is exhausted.")
```

Comment:
I'm not sure this second option is particularly actionable. As a user, I know what frames being zero length means, but what does 'generator is exhausted' mean? Maybe if it said something about the animation being finished.

Reply:
By the time we get here we no longer know what the user actually passed us (we have wrapped it with a callable that we call to get a iterable).  I'll have a think about how to make this clearer.

Reply:
"There are no more frames or the animation generation has stopped creating frames." ?